pymatgen/entries/computed_entries.py

```diff
@@ -12,25 +12,78 @@
 
 import json
 
-from monty.json import MontyEncoder, MontyDecoder
+from monty.json import MontyEncoder, MontyDecoder, MSONable
 
 from pymatgen.core.composition import Composition
 from pymatgen.core.structure import Structure
 from pymatgen.entries import Entry
 
-__author__ = "Shyue Ping Ong, Anubhav Jain"
+__author__ = "Ryan Kingsbury, Shyue Ping Ong, Anubhav Jain"
 __copyright__ = "Copyright 2011, The Materials Project"
 __version__ = "1.1"
 __maintainer__ = "Shyue Ping Ong"
 __email__ = "shyuep@gmail.com"
 __status__ = "Production"
-__date__ = "Apr 30, 2012"
+__date__ = "April 2020"
+
+
+class EnergyAdjustment(MSONable):
+    """
+    Lightweight class to contain information about an energy adjustment or
+    energy correction.
+    """
+    def __init__(self, value, name="Manual adjustment", cls="None", description=""):
+        """
+        Args:
+            value: float, value of the energy adjustment in eV
+            name: str, human-readable name of the energy adjustment.
+                (Default: Manual adjustment)
+            cls: str, the name of the Compatibility class used to generate the
+                energy adjustment. (Default: None)
+            description: str, human-readable explanation of the energy adjustment.
+        """
+        self.value = value
+        self.name = name
+        self.cls = cls
+        self.description = description
+
+    def __repr__(self):
+        output = ["{}:".format(self.__class__.__name__),
+                  "  Name: {}".format(self.name),
+                  "  Value: {:.3f} eV".format(self.value),
+                  "  Description: {}".format(self.description),
+                  "  Generated by: {}".format(self.cls.get("@class", self.cls))]
+        return "\n".join(output)
+
+
+class ConstantEnergyAdjustment(EnergyAdjustment):
+    """
+    A constant energy adjustment applied to a ComputedEntry. Useful in energy referencing
+    schemes such as the Aqueous energy referencing scheme.
+    """
+    pass
+
+
+class CompositionEnergyAdjustment(EnergyAdjustment):
+    """
+    An energy adjustment applied to a ComputedEntry based on the atomic composition.
+    Used in various DFT energy correction schemes.
+    """
+    pass
+
+
+class TempEnergyAdjustment(EnergyAdjustment):
+    """
+    An energy adjustment applied to a ComputedEntry based on the temperature.
+    Used, for example, to add entropy to DFT energies.
+    """
+    pass
```

Comment:
@shyamd I've put a layer of abstraction in place for `EnergyAdjustmetnt` classes, but currently the child classes don't do anything. What did you have in mind for the different types of `EnergyAdjustment`?

I'm thinking we could make `.value` an abstract method/property, then define it as follows:

`ConstantEnergyAdjustment(value)` -> value = value (no change from base class)

`CompositionEnergyAdjustment(adj_per_atom, comp)` -> value = adjustment per atom * no. of atoms

`TempEnergyAdjustment(adj_per_deg, temp)` -> value = adjustment per degree * temperature in K

What are your thoughts?

Reply:
Yeah, that was definitely my idea.

I would also add `ManualAdjustment` as just a child of `ConstantAdjustment` with an updated docstring. This was you can filter it out.